addons/stock/models/product.py

```diff
@@ -164,33 +165,37 @@ def _get_domain_locations(self):
                 [('location_id.company_id', '=', False), ('location_dest_id.company_id', '=', company_id)],
                 [('location_id.company_id', '=', company_id), ('location_dest_id.company_id', '=', False),
             ])
-        location_ids = []
-        if self.env.context.get('location', False):
-            if isinstance(self.env.context['location'], int):
-                location_ids = [self.env.context['location']]
-            elif isinstance(self.env.context['location'], str):
-                domain = [('complete_name', 'ilike', self.env.context['location'])]
-                if self.env.context.get('force_company', False):
-                    domain += [('company_id', '=', self.env.context['force_company'])]
-                location_ids = self.env['stock.location'].search(domain).ids
+
+        def _search_ids(model, value, force_company_id):
```

Comment:
add a comment: dummy field uset in the search view will always return a list of ids or a list of string